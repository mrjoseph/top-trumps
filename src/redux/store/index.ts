import { combineReducers, createStore, applyMiddleware } from 'redux';
import { gameReducer } from '../reducers/reducers';
import createSagaMiddleware from 'redux-saga';
import mySaga from '../saga/sagas';

const sagaMiddleware = createSagaMiddleware()

const routeReducer = combineReducers({
    gameReducer
  });

export type AppState = ReturnType<typeof routeReducer>

const middleware = [sagaMiddleware];

const store = createStore(routeReducer, applyMiddleware(...middleware));
sagaMiddleware.run(mySaga);  
export default store;
