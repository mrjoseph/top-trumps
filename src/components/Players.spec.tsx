import React from 'react';
import ActivePlayers, { IActivePlayer } from './Players';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';

describe('ActivePlayers', () => {
    it('renders correctly', () => {
        const props:IActivePlayer = {
            numberOfPlayers: 0,
            starwars: []
        }
        const tree = renderer.create(
            <Router>
            <Provider store={store}>
                <ActivePlayers  {...props}/>
            </Provider>
            </Router>).toJSON();
        expect(tree).toMatchSnapshot();
      });
});