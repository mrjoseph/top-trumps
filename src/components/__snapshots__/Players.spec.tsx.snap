// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivePlayers renders correctly 1`] = `
<ul
  className="sc-bwzfXH fLjpAL"
/>
`;
